runtimes/pythonrt/runner/main.py

```diff
@@ -444,21 +444,25 @@ def on_event(self, fn, event):
             runner_id=self.id,
         )
 
-        if result.error:
-            req.error = result_error(result.error)
-            tb = pb_traceback(result.traceback)
-            req.traceback.extend(tb)
-        else:
-            try:
-                data = pickle.dumps(result)
-                req.result.custom.data = data
-                req.result.custom.value.CopyFrom(values.safe_wrap(result.value))
-            except (TypeError, pickle.PickleError) as err:
-                req.error = f"can't pickle {result.value} - {err}"
+        try:
+            if result.error:
+                req.error = result_error(result.error)
+                tb = pb_traceback(result.traceback)
+                req.traceback.extend(tb)
+            else:
+                try:
+                    data = pickle.dumps(result)
+                    req.result.custom.data = data
+                    req.result.custom.value.CopyFrom(values.safe_wrap(result.value))
+                except (TypeError, pickle.PickleError) as err:
+                    req.error = f"can't pickle {result.value} - {err}"
```

Comment:
Nested `try` are pointless and very inefficient: move the inner `except` to the same level of the new one (make sure `except Exception` is always last)

```suggestion
        try:
            if result.error:
                req.error = result_error(result.error)
                tb = pb_traceback(result.traceback)
                req.traceback.extend(tb)
            else:
                data = pickle.dumps(result)
                req.result.custom.data = data
                req.result.custom.value.CopyFrom(values.safe_wrap(result.value))
        except (TypeError, pickle.PickleError) as err:
            req.error = f"can't pickle {result.value} - {err}"
```

Reply:
technically since this will run only once and end I don't really mind
left it with nesting since the error handling was like that before

but sure, refactored